assembly/bindgen.ts

```diff
@@ -110,10 +110,27 @@ function encode<T, Output = Uint8Array>(value: T, name: string | null = "", enco
       } else { // Is an object
         if (value instanceof u128) {
           encoder.setString(name, value.toString());
+        } else if (value instanceof Map) {
+          //@ts-ignore
+          assert(nameof<indexof<T>>() == "String", "Can only encode maps with string keys");
```

Comment:
Can we see if the key is easily coerceable to a string instead? Something like:

```suggestion
          assert(nameof<indexof<T>>() == "String" || T.toString, "Can only encode maps with string keys");
```

Reply:
`T` here is the `Map<K,V>`, so this is checking that `K == string`.  

Perhaps I could try:
```ts
let key: indexof<T>;

if (isDefined(key.toString) {
....
```

But I fear this might open a can of worms when trying to decode.  So I think we should just make a tracking issue and return to this if there is demand for it. We should also look at what rust does.

Reply:
Good point on the decoding; we can leave as-is for now. Can you make that tracking issue?